src/streaming/models/LowLatencyThroughputModel.js

```diff
@@ -0,0 +1,184 @@
+ *  contributors may be used to endorse or promote products derived from this software
+ *  without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS AS IS AND ANY
+ *  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ *  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ *  IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ *  INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ *  NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ *  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ *  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+import FactoryMaker from '../../core/FactoryMaker';
+
+function LowLatencyThroughputModel() {
+
+    const LLTM_MAX_MEASUREMENTS = 10;
+    // factor (<1) is used to reduce the real needed download time when at very bleeding live edge
+    const LLTM_SEMI_OPTIMISTIC_ESTIMATE_FACTOR = 0.7;
+
+    let dashMetrics;
+
+    let instance;
+    let measurements = {};
+
+    /**
+     *
+     * @param {*} config
+     */
+    function setConfig(config) {
+        dashMetrics = config.dashMetrics;
+    }
+
+    /**
+     *
+     * @param {*} request
+     * @returns
+     */
+    function getEstimatedDownloadDurationMS(request) {
+        const lastMeasurement = measurements[request.mediaType].slice(-1).pop();
+
+        // fetch duration was longer than segment duration? 
+        if (lastMeasurement.segDurationMS < lastMeasurement.fetchDownloadDurationMS) {
+            return lastMeasurement.fetchDownloadDurationMS;
+        }
+
+        // we have requested a fully available segment -> most accurate throughput calculation
+        // this usually happens at startup
+        // ... and if requests are delayed artificially
+        if (lastMeasurement.ast <= lastMeasurement.requestTime - lastMeasurement.segDurationMS) {
+            return lastMeasurement.fetchDownloadDurationMS;
+        }
+
+        // get all chunks that have been downloaded before fetch reached bleeding live edge
+        // the remaining chunks loaded at production rate we will approximated
+        const chunkAvailabePeriod = lastMeasurement.requestTime - lastMeasurement.ast;
+        let chunkBytesBBLE = 0; // BBLE -> Before bleeding live edge
+        let chunkDownloadtimeMSBBLE = 0;
+        let chunkCount = 0;
+        for (let index = 0; index < lastMeasurement.measurement.length; index++) {
+            const chunk = lastMeasurement.measurement[index];
+            if (chunkAvailabePeriod < chunkDownloadtimeMSBBLE + chunk.dur) {
+                break;
+            }
+            chunkDownloadtimeMSBBLE += chunk.dur;
```

Comment:
`chunk.dur` is the download time for the data chunk and not the duration of the chunk right?